portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,93 @@
 
 package com.google.sps.servlets;
 
+import com.google.gson.Gson;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+/** Servlet that returns some example content. */
+@WebServlet("/leave-comment")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Yijia!</h1>");
-  }
+    ArrayList<Comment> messages = new ArrayList<Comment>();
+    int counter = -1;
```

Comment:
Is this counter variable required? Would your code work without it?

Reply:
I use this variable to get around not being able to pass in user input to doGet. So whenever user enters a number (of comments to display), it first calls doPost, and doPost update this counter variable. Then the page calls doGet, which now has access to this counter variable to update the corresponding comments to display. I saw this method in the technical support chat, but I am not sure if there is a better way to do it.